packages/riipen-ui/src/components/editor/Editor.jsx

```diff
@@ -0,0 +1,638 @@
+import { convertFromHTML, convertToHTML } from "draft-convert";
+import {
+  AtomicBlockUtils,
+  CompositeDecorator,
+  Editor as DraftJsEditor,
+  EditorState,
+  getDefaultKeyBinding,
+  Modifier,
+  RichUtils,
+  SelectionState
+} from "draft-js";
+import PropTypes from "prop-types";
+import React from "react";
+import css from "styled-jsx/css";
+
+import ThemeContext from "../../styles/ThemeContext";
+
+import BlockStyleControls from "./BlockStyleControls";
+import InlineStyleControls from "./InlineStyleControls";
+import RteImage from "./Image";
+
+// draft-convert configuration for converting HTML to EditorState.
+export const fromHtmlConfig = {
+  htmlToEntity: (nodeName, node, createEntity) => {
+    switch (nodeName) {
+      case "a":
+        return createEntity("LINK", "MUTABLE", { url: node.href });
+      case "img":
+        return createEntity("IMAGE", "IMMUTABLE", {
+          alt: node.alt,
+          src: node.src
+        });
+      default:
+        return null;
+    }
+  },
+  htmlToBlock: nodeName => {
+    switch (nodeName) {
+      case "div":
+        return {
+          type: "atomic",
+          data: {}
+        };
+      default:
+        return null;
+    }
+  }
+};
+
+// draft-convert configuration for converting EditorState to HTML.
+const toHtmlConfig = {
+  blockToHTML: block => {
+    switch (block.type) {
+      case "code-block":
+        return <pre />;
+      case "atomic":
+        return <div />;
+      default:
+        return null;
+    }
+  },
+  entityToHTML: (entity, originalText) => {
+    switch (entity.type) {
+      case "LINK":
+        return <a href={entity.data.url}>{originalText}</a>;
+
+      case "IMAGE":
+        return <img alt={entity.data.alt} src={entity.data.src} />;
+      default:
+        return originalText;
+    }
+  }
+};
+
+// draft-js map of inline style types to the corresponding style displayed in the editor.
+const styleMap = {
+  CODE: {
+    backgroundColor: "rgba(0, 0, 0, 0.05)",
+    fontFamily: '"Inconsolata", "Menlo", "Consolas", monospace',
+    padding: 2
+  }
+};
+
+const getBlockComponent = block => {
+  switch (block.getType()) {
+    case "atomic":
+      return { component: RteImage, editable: false };
+    default:
+      return null;
+  }
+};
+
+class Editor extends React.Component {
+  static propTypes = {
+    /**
+     * Additional controls to display.
+     */
+    additionalControls: PropTypes.node,
+
+    /**
+     * The id of the element to use as the aria-label for the Editor.
+     */
+    ariaLabelledBy: PropTypes.string,
+
+    /**
+     * Optional array of whitelisted styles
+     * ex. ['code-block', 'BOLD', 'LINK']
+     */
+    controlWhitelist: PropTypes.arrayOf(PropTypes.string),
+
+    /**
+     * Position of controls for the editor
+     */
+    controlPosition: PropTypes.oneOf(["top", "bottom"]),
+
+    /**
+     * The decorator for the editor.
+     */
+    decorator: PropTypes.shape({ type: PropTypes.oneOf([CompositeDecorator]) }),
```

Comment:
Moved the decorator out so that devs need to pass custom decorators.  The only place this makes a difference so far is with links, but we'll pass a decorator in so should be no problemo.